recipes/stx/all/conanfile.py

```diff
@@ -0,0 +1,147 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class STXConan(ConanFile):
+    name = 'stx'
+    homepage = 'https://github.com/lamarrr/STX'
+    license = 'MIT'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = 'C++17 & C++ 20 error-handling and utility extensions.'
+    generators = 'cmake', 'cmake_find_package'
+    topics = 'error-handling', 'result', 'option', 'backtrace', 'panic'
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'backtrace': [True, False],
+        'panic_backtrace': [True, False],
+        'override_panic_handler': [True, False],
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'visible_panic_hook': [True, False],
+    }
+    default_options = {
+        'backtrace': False,
+        'override_panic_handler': False,
+        'shared': False,
+        'fPIC': True,
+    }
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.panic_backtrace == None:
+            self.options.panic_backtrace = (
+                self.options.backtrace and
+                not self.options.override_panic_handler
+            )
+
+        if self.options.visible_panic_hook == None:
+            self.options.visible_panic_hook = self.options.shared
```

Comment:
Perhaps this could be easier to consume if the options were an array.. Ex:
```py
'backtrace': [None, 'panic', 'override_handler'],
```

Reply:
The backtrace option controls whether backtrace functionality in general is available,
panic_backtrace enables using/showing backtraces in the default panic handler, which requires backtrace to be on,
override_panic_handler turns off the default panic handler.
So panic_backtrace and override_panic_handler  are mutually exclusive, but backtrace and override_panic_handler can be set independently.
I tried to expose the same major options: https://github.com/lamarrr/STX#cmake-configuration-options

Reply:
Hmm, I see what you are saying. I am not the expert on options, we'll see if anyone else has a better solution but looking at this for the first time it's not obvious (which is partly the fault of upstream for making it so confusing)... I am not sure there's any other recipe that omits the default and release on None as an "auto" behavior.

Reply:
Maybe like this:
```py
'backtrace': [True, False],
'panic_handler': [None, 'default', 'backtrace']
```
and then reject the `backtrace=False, panic_handler='backtrace'` combination with a ConanInvalidConfiguration?

Reply:
I think that would cover all the combinations, while being easier to understand:
None - no panic_handler defined, you get to define it
default - you get the default panic handler
backtrace - you get a panic handler that also prints a backtrace

Reply:
That sounds much better!